src/prefect/futures.py

```diff
@@ -104,18 +106,23 @@ class PrefectFuture(Generic[R, A]):
 
     def __init__(
         self,
-        task_run: TaskRun,
-        run_key: str,
+        name: str,
+        key: UUID,
         task_runner: "BaseTaskRunner",
         asynchronous: A = True,
         _final_state: State[R] = None,  # Exposed for testing
     ) -> None:
-        self.task_run = task_run
-        self.run_key = run_key
+        self.key = key
+        self.name = name
         self.asynchronous = asynchronous
+        self.task_run = None
         self._final_state = _final_state
         self._exception: Optional[Exception] = None
         self._task_runner = task_runner
+        self._submitted = anyio.Event()
+        import asyncio
```

Comment:
Why avoid the import until here? Might be worth a comment.

Reply:
This is left over from my sketch, should be moved I think.